server/api/dataset.py

```diff
@@ -252,51 +263,36 @@ def submitReviewImages(self, dataset, params):
     @describeRoute(
         Description('Register metadata with a dataset.')
         .param('id', 'The ID of the dataset.', paramType='path')
-        .param('uploadFolderId', 'The ID of the folder that contains metadata '
-               'in a .csv file.')
+        .param('metadataFileId', 'The ID of the .csv metadata file.')
     )
     @access.user
-    @loadmodel(model='dataset', plugin='isic_archive', level=AccessType.READ)
+    @loadmodel(model='dataset', plugin='isic_archive', force=True)
```

Comment:
Combined with the `User.requireCreateDataset(user)` requirement below, I think this should be enough access checking. I don't think we can expect that all metadata uploaders will have `WRITE` access, since we may want to quickly take away end users' ability to delete items or arbitrarily modify metadata directly.